src/test/java/core/basesyntax/servise/impl/FruitTransactionProcessorServiceImplTest.java

```diff
@@ -0,0 +1,153 @@
+package core.basesyntax.servise.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.servise.FruitTransactionProcessorService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.operation.BalanceOperation;
+import core.basesyntax.strategy.operation.OperationHandler;
+import core.basesyntax.strategy.operation.PurchaseOperation;
+import core.basesyntax.strategy.operation.ReturnOperation;
+import core.basesyntax.strategy.operation.SupplyOperation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionProcessorServiceImplTest {
+    private static FruitTransactionProcessorService processorService;
+    private static List<FruitTransaction> fruitTransactions;
+
+    @BeforeClass
+    public static void beforeClass() {
+        Map<FruitTransaction.Operation, OperationHandler> handlerMap = new HashMap<>();
+        handlerMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation());
+        handlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperation());
+        handlerMap.put(FruitTransaction.Operation.RETURN, new ReturnOperation());
+        handlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(handlerMap);
+        processorService = new FruitTransactionProcessorServiceImpl(operationStrategy);
+        fruitTransactions = new ArrayList<>();
+    }
+
+    @After
+    public void after() {
+        Storage.fruitStorage.clear();
+        fruitTransactions.clear();
+    }
+
+    @Test
+    public void process_validFruitTransactions_Ok() {
```

Comment:
too huge method, let's create a separate private method that will be creating FruitTransaction and add it to DB, in this method you should pass parameters needed for the creation of FruitTransaction